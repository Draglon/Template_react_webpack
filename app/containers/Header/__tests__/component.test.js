import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HeaderComponent from '../component';

describe('<HeaderComponent /> component', () => {
  let props = {
    profile: { avatar: 'gravatar_hash', name: 'name', username: 'username' },
    removeSessionId: jest.fn(),
  };
  let component;

  it('Render property - with avatar', () => {
    component = shallow(<HeaderComponent {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render property - without avatar', () => {
    props = {
      ...props,
      profile: { avatar: '', name: 'name', username: 'username' },
    };
    component = shallow(<HeaderComponent {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
