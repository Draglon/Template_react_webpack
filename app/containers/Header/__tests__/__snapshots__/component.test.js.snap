// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> component Render property 1`] = `
<Adapter>
  <Row
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col>
      <Text
        className="logo"
      >
        THE MOVIE DB
      </Text>
    </Col>
    <Col>
      <Avatar
        icon="user"
        shape="circle"
        size="default"
      />
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={
          <Menu>
            <MenuItem>
              <Link
                replace={false}
                to="/stubs/dashboard"
              >
                Dashboard
              </Link>
            </MenuItem>
            <Divider
              className=""
              disabled={true}
              style={Object {}}
            />
            <MenuItem>
              <Link
                replace={false}
                to="/stubs/lists"
              >
                My Lists
              </Link>
            </MenuItem>
            <MenuItem>
              <Link
                replace={false}
                to="/stubs/watchlist"
              >
                Watchlist
              </Link>
            </MenuItem>
            <MenuItem>
              <Link
                replace={false}
                to="/stubs/favorites"
              >
                Favorites
              </Link>
            </MenuItem>
            <Divider
              className=""
              disabled={true}
              style={Object {}}
            />
            <MenuItem
              onClick={[MockFunction]}
            >
              Logout
            </MenuItem>
          </Menu>
        }
        placement="bottomLeft"
      >
        <Text>
          <Icon
            type="caret-down"
          />
        </Text>
      </Dropdown>
    </Col>
  </Row>
</Adapter>
`;
